### ISSUE TITLE
Fatal error attempting to view Template parts using gbcptedit

### ISSUE THREAD

**Issue by bobbingwide** [bobbingwide/gbcptedit#3]
Fatal error visiting https://seriouslybonkers.com/wp-admin/edit.php?post_type=wp_template_part

Fatal error: Uncaught ArgumentCountError: 3 arguments are required, 2 given in 
```
/home/customer/www/seriouslybonkers.com/public_html/wp-includes/link-template.php:1580 Stack trace: #0 
/home/customer/www/seriouslybonkers.com/public_html/wp-includes/link-template.php(1580): sprintf('site-editor.php...', 7156) bobbingwide/bobbingwide#1 
/home/customer/www/seriouslybonkers.com/public_html/wp-admin/includes/class-wp-posts-list-table.php(1515): get_delete_post_link(Object(WP_Post)) bobbingwide/bobbingwide#2
 /home/customer/www/seriouslybonkers.com/public_html/wp-admin/includes/class-wp-posts-list-table.php(1073): WP_Posts_List_Table->handle_row_actions(Object(WP_Post), 'title', 'title') bobbingwide/bobbingwide#3
 /home/customer/www/seriouslybonkers.com/public_html/wp-admin/includes/class-wp-list-table.php(1787): WP_Posts_List_Table->_column_title(Object(WP_Post), 'title column-ti...', 'data-colname="T...', 'title') bobbingwide/bobbingwide#4
 /home/customer/www/seriouslybonkers.com/public_html/wp-admin/includes/class-wp-posts-list-table.php(1432): WP_List_Table->single_row_columns(Object(WP_Post)) bobbingwide/bobbingwide#5 
/home/customer/www/seriouslybonkers.com/public_html/wp-admin/includes/class-wp-posts-list-table.php(834): WP_Posts_List_Table->single_row(Object(WP_Post), 0) bobbingwide/bobbingwide#6 
/home/customer/www/seriouslybonkers.com/public_html/wp-admin/includes/class-wp-posts-list-table.php(810): WP_Posts_List_Table->_display_rows(Array, 0) bobbingwide/bobbingwide#7 
/home/customer/www/seriouslybonkers.com/public_html/wp-admin/includes/class-wp-list-table.php(1706): WP_Posts_List_Table->display_rows() bobbingwide/bobbingwide#8 
/home/customer/www/seriouslybonkers.com/public_html/wp-admin/includes/class-wp-list-table.php(1633): WP_List_Table->display_rows_or_placeholder() bobbingwide/fizzie#52 
/home/customer/www/seriouslybonkers.com/public_html/wp-admin/edit.php(503): WP_List_Table->display() bobbingwide/bobbingwide#10 
{main} thrown in /home/customer/www/seriouslybonkers.com/public_html/wp-includes/link-template.php on line 1580
```

**Comment by bobbingwide** [comment]
The problem occurs because sprintf() is expecting 3 parameters but is only being passed 2.
The `_edit_link` for post types which are edited using the site editor is
```
site-editor.php?postType=%s&postId=%s&canvas=edit
 ```
so sprintf() expects two additional parameters for the `postType` and `postId`.
get_delete_post_link() is only passed the post ID since it expects the link to be
```
post.php?post=%d
```

Unfortunately `$_edit_link` is documented as being:

FOR INTERNAL USE ONLY! 
URL segment to use for edit link of this post type. Default 'post.php?post=%d'.

**Comment by bobbingwide** [comment]
Problem first occurs in WordPress 6.3. Fails with PHP 8.1 - 8.3
It's OK with WordPress 6.2 for PHP 8.1 - 8.3

**Comment by bobbingwide** [comment]
I've raised a TRAC against `get_delete_post_link()`. 
See https://core.trac.wordpress.org/ticket/61716

I've also developed a workaround to prevent the Fatal error. 
This is implemented as code that disables the `delete_post` capability.
The admin pages are now displayed.

Problems still exist though...

- The workaround doesn't check the URL so `delete post` is disabled across the board.
- The Edit links invoke the Site editor not the standard block editor.
